src/sentry/monitors/validators.py

```diff
@@ -527,3 +536,137 @@ def validate_ids(self, value):
         ).count() != len(value):
             raise ValidationError("Not all ids are valid for this organization.")
         return value
+
+
+class MonitorDataSourceValidator(BaseDataSourceValidator[Monitor]):
```

Comment:
the naming here coulllld get a bit confusing -- since `Detector`s are called `monitors` in the product.

Reply:
I can call this crons instead, although it's generally confusing because our model is called `Monitor`, lol